test/functional/test-document-click.js

```diff
@@ -197,4 +199,46 @@ describe('test-document-click onDocumentElementClick_', () => {
       expect(replaceStateSpy.args[0][2]).to.equal('#test');
     });
   });
+
+  describe('when linking to custom protocols e.g. whatsapp:', () => {
+    let sandbox;
+    beforeEach(() => {
+      sandbox = sinon.sandbox.create();
+      win.parent = {};
+      win.top = {
+        location: {
+          href: 'https://google.com'
+        }
+      };
+      tgt.href = 'whatsapp://send?text=hello';
+    });
+
+    afterEach(() => {
+      sandbox.restore();
+      sandbox = null;
+    });
+
+    it('should set top.location.href on Safari iOS when embedded', () => {
+      sandbox.stub(platform, 'isIos').returns(true);
+      sandbox.stub(platform, 'isSafari').returns(true);
+      onDocumentElementClick_(evt, viewport);
+      expect(win.top.location.href).to.equal('whatsapp://send?text=hello');
+      expect(preventDefaultSpy.callCount).to.equal(1);
+    });
+
+    it('should not do anything on other platforms when embedded', () => {
+      sandbox.stub(platform, 'isIos').returns(true);
+      sandbox.stub(platform, 'isSafari').returns(false);
```

Comment:
This is testing Chrome on iOS, not other platforms. Can you add another test for Android?